securityfilter.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/shurcooL/githubv4"
+)
+
+func filterSecurityPullRequests(
+	client *githubv4.Client,
+	pullRequests *[]pullRequest,
+) ([]pullRequest, error) {
+	filteredPullRequests := []pullRequest{}
+	for _, pr := range *pullRequests {
+		type vulnQuery struct {
+			Repository struct {
+				VulnerabilityAlerts struct {
+					Nodes []struct {
+						VulnerableRequirements string
+						State                  string
+						SecurityVulnerability  struct {
+							Package struct {
+								Name string
+							}
+						}
+					}
+				} `graphql:"vulnerabilityAlerts(first:100,states:OPEN)"`
+			} `graphql:"repository(owner: $owner, name: $name)"`
+		}
+		vulnVars := map[string]interface{}{
+			"owner": githubv4.String(pr.owner),
+			"name":  githubv4.String(pr.repository),
+		}
+		var vulnQ vulnQuery
+
+		if err := client.Query(context.Background(), &vulnQ, vulnVars); err != nil {
```

Comment:
pass a context down to this function rather than using the default context - on the level above it should be possible to give `cmd.Context()`

Reply:
Fixed!